src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -769,6 +771,252 @@ protected bool IsAlive
+        {
+            Autodesk.Revit.DB.Element parent;
+            var stairElement = element as Autodesk.Revit.DB.Architecture.StairsLanding;
+            if (stairElement == null)
+                throw new InvalidOperationException(Properties.Resources.NoParentElement);
+            parent = stairElement.GetStairs();
+            return parent;
+        }
+
+        /// <summary>
+        /// Transforms this Element from source CoordinateSystem to new context CoordinateSystem.
+        /// </summary>
+        /// <param name="fromCoordinateSystem">Source CoordinatSystem</param>
+        /// <param name="contextCoordinateSystem">Context CordinateSystem</param>
+        /// <returns>Transformed Element</returns>
+        public Element Transform(CoordinateSystem fromCoordinateSystem, CoordinateSystem contextCoordinateSystem)
+        {
+            TransactionManager.Instance.EnsureInTransaction(Document);
+            var transManager = TransactionManager.Instance.TransactionWrapper;
+            var t = transManager.StartTransaction(Document);
+            try
+            {
+                SetLocationFromCS(this, fromCoordinateSystem, contextCoordinateSystem);
+                RotateElementFromCS(this, fromCoordinateSystem, contextCoordinateSystem);
+                t.CommitTransaction();
+            }
+            catch (Exception ex)
+            {
+                t.CancelTransaction();
+                throw new Exception(ex.Message);
+            }         
+            TransactionManager.Instance.TransactionTaskDone();
+            return this;
+        }
+
+        private void RotateElementFromCS(Element element, CoordinateSystem fromCS, CoordinateSystem contextCS)
+        {
+            var familyInstance = element as FamilyInstance;
+            if (familyInstance == null)
+                throw new NullReferenceException(nameof(element));
+            var newTransform = contextCS.ToTransform() as Autodesk.Revit.DB.Transform;
+
+            var oldTransform = fromCS.ToTransform() as Autodesk.Revit.DB.Transform;
+            double[] oldRotationAngles;
+            TransformUtils.ExtractEularAnglesFromTransform(oldTransform, out oldRotationAngles);
+            double[] newRotationAngles;
+            TransformUtils.ExtractEularAnglesFromTransform(newTransform, out newRotationAngles);
+            // Convert Eular angle to degrees
+            var rotationDegrees = (newRotationAngles.FirstOrDefault() / (2 * Math.PI)) * 360;
+            var newRotationAngle = rotationDegrees * Math.PI / 180;
+
+            if (!oldRotationAngles[0].AlmostEquals(newRotationAngle, 1.0e-6))
+            {
+                var rotateAngle = newRotationAngle - oldRotationAngles[0];
+                var axis = Autodesk.Revit.DB.Line.CreateUnbound(oldTransform.Origin, oldTransform.BasisZ);
+                Autodesk.Revit.DB.ElementTransformUtils.RotateElement(Document, new Autodesk.Revit.DB.ElementId(Id), axis, -rotateAngle);
```

Comment:
hmm, will this always be negative?

Reply:
using the code from the original SeRotation method which always has the `rotateAngle` as negative.